### PR TITLE
Feature/dependency injection

### PR DESCRIPTION
**Pull request by VIbanichuka**
I have tried to make the application a little bit cleaner.

### CODE REVIEW

File: src/SimpleCalculator.Console/UserConsoleInputValidation.cs [hunk 1]
```diff
@@ -1,4 +1,5 @@
-namespace Calculations;
+using Calculations;
+namespace SimpleCalculator;
 
 public class UserConsoleInputValidation : IUserConsoleInputValidation
```
Comment (NMoskalev) [inline src/SimpleCalculator.Console/UserConsoleInputValidation.cs:4]: It makes sense to rename `IUserConsoleInputValidation` interface

File: src/Calculations/Addition.cs [hunk 2]
```diff
@@ -12,16 +11,21 @@ public Addition(IUserConsoleInputValidation validateInput)
     public double FirstNumber { get; private set; }
     public double SecondNumber { get; private set; }
 
-    public void AddWorkFlow()
+    public double Num1()
```
Comment (NMoskalev) [inline src/Calculations/Addition.cs:14]: it's not clear from a name what `Num1` method is doing. Method name must have a verb or action

File: src/Calculations/Addition.cs [hunk 3]
```diff
@@ -12,16 +11,21 @@ public Addition(IUserConsoleInputValidation validateInput)
...
+    {
+        return FirstNumber = _validateInput.ValidateInput();
+    }
+
+    public double Num2()
+    {
+        return SecondNumber = _validateInput.ValidateInput();
```
Comment (NMoskalev) [inline src/Calculations/Addition.cs:21]: why wouldn't do a validation outside of the class? In this case we won't need `Num1`, `Num2`, `DoAdd` methods

**Review by NMoskalev — COMMENTED** [review]
I like DI implementation! I've added a couple of comments, could you please take a look?

File: src/SimpleCalculator.Console/UserConsoleInputValidation.cs (hunk 1 above)
Comment (VIbanichuka) [reply src/SimpleCalculator.Console/UserConsoleInputValidation.cs:4]: Oh, absolutely. My bad!

**Pull request merged**